resources-domain/resources-import-producer/tests/unit/transformExternalResourceToApiResource.test.ts

```diff
@@ -0,0 +1,898 @@
+    id: partialResource.id || '',
+    updatedAt: partialResource.updatedAt || '',
+    name: partialResource.name || '',
+    attributes: {
+      ResourceStringAttributes: partialResource.attributes?.ResourceStringAttributes || [],
+      ResourceNumberAttributes: partialResource.attributes?.ResourceNumberAttributes || [],
+      ResourceBooleanAttributes: partialResource.attributes?.ResourceBooleanAttributes || [],
+      ResourceDateTimeAttributes: partialResource.attributes?.ResourceDateTimeAttributes || [],
+      ResourceReferenceStringAttributes:
+        partialResource.attributes?.ResourceReferenceStringAttributes || [],
+    },
+  },
+});
+
+describe('transformExternalResourceToApiResource - dynamic captures', () => {
+  test('Dynamic capture only - captures all the keys', async () => {
+    const resource = {
+      attribute: {
+        key1: 'value 1',
+        key2: 'value 2',
+      },
+    };
+
+    const mapping: MappingNode = {
+      attribute: {
+        children: {
+          '{key}': attributeMapping('ResourceStringAttributes', '{key}'),
+        },
+      },
+    };
+
+    const expected: ImportApiResource = mergeWithCleanResource({
+      attributes: {
+        ResourceStringAttributes: [
+          {
+            key: 'key1',
+            value: 'value 1',
+            info: null,
+            language: null,
+          },
+          {
+            key: 'key2',
+            value: 'value 2',
+            info: null,
+            language: null,
+          },
+        ],
+      },
+    });
+
+    const result = transformExternalResourceToApiResource(mapping, resource);
+    expect(result).toMatchObject(expected);
+  });
+
+  test('Dynamic capture and static mapping - captures only the non-static keys', async () => {
+    const resource = {
+      attribute: {
+        another: true,
+        key1: 'value 1',
+        key2: 'value 2',
+      },
+    };
+
+    const mapping: MappingNode = {
+      attribute: {
+        children: {
+          another: attributeMapping('ResourceBooleanAttributes', 'another'),
+          '{key}': attributeMapping('ResourceStringAttributes', '{key}'),
+        },
+      },
+    };
+
+    const expected: ImportApiResource = mergeWithCleanResource({
+      attributes: {
+        ResourceStringAttributes: [
+          {
+            key: 'key1',
+            value: 'value 1',
+            info: null,
+            language: null,
+          },
+          {
+            key: 'key2',
+            value: 'value 2',
+            info: null,
+            language: null,
+          },
+        ],
+        ResourceBooleanAttributes: [
+          {
+            key: 'another',
+            value: true,
+            info: null,
+          },
+        ],
+      },
+    });
+
+    const result = transformExternalResourceToApiResource(mapping, resource);
+    expect(result).toMatchObject(expected);
+  });
+});
+
+describe('transformExternalResourceToApiResource - Simple mapping, flat structure', () => {
```

Comment:
Can we try to use an Arrange Act Assert type naming for tests here in line with our standards, I appreciate in might make the tests a little more verbose but I think we'll appreciate the clarity dowen the road when nobody has looked at this code for 6 month :-P

Also - it looks like you cover most of the mapping functionality, but perhaps some thinks like custom mappers for various attributes could have their own specific tests for that functionality, again, just to be super explicit